Bug: fix travis deploy script

andreas-bok-sociomantic opened the pull request:
In spite of no build error the deploy step 
didn't succeed. 

Travis wraps the condition inside double brackets.
Thus && is expected as comparison operator

andreas-bok-sociomantic commented on .travis.yml:
```diff
@@ -72,11 +72,8 @@ script:
 deploy:
     provider: script
-    script: beaver bintray upload -N pkg/deb/*.deb
+    script: beaver bintray upload -N deb/*.deb
```
Should the `upload` script be called with the full path? 
Travis errors with this for the deploy step:  

```
Preparing deploy

Deploying application
sh: 1: beaver: not found

Script failed with status 127
failed to deploy
```

leandro-lucarella-sociomantic replied:
Ahhh, right! That would be one choice, but it's better to add it to the path:

```ỳml
env:
    global:
        - PATH="$(git config -f .gitmodules submodule.beaver.path)/bin:$PATH"
```

leandro-lucarella-sociomantic commented on .travis.yml:
```diff
@@ -72,11 +72,8 @@ script:
 deploy:
     provider: script
-    script: beaver bintray upload -N pkg/deb/*.deb
+    script: beaver bintray upload -N deb/*.deb
```
BTW, are you sure this should be `deb/*.deb` and not `pkg/deb/*.deb`?

andreas-bok-sociomantic replied:
Yeah, the commit, https://github.com/sociomantic-tsunami/libdrizzle-redux/commit/b708fd9e263d559332a9fda736f22fe6c6cd3f4c,  which moves the packing scripts to the `pkg` is on the 5.x.x. MIssed that when reviewing after the rebase on v5.5.x

leandro-lucarella-sociomantic replied:
Ahh, ok, sorry.

leandro-lucarella-sociomantic commented on .travis.yml:
```diff
@@ -72,11 +72,8 @@ script:
-            "$TRAVIS_OS_NAME" = "linux" -a
-            "$CC" = "gcc"
-
+        condition: $TRAVIS_OS_NAME = 'linux' && $CC = 'gcc'
```
And do you know this is really a problem? Because it should work. I mean, putting it in one line seems good if it fits but the `-a` -> `&&` shouldn't be necessary if the condition is supposed to be pasted in at `if [[ ... ]]`.

andreas-bok-sociomantic replied:
Ah. So is it quoting of the conditions which maybe messes up the if statement? 
The parsing error for the `-a` syntax is : 

```
[0K/home/travis/.travis/job_stages: line 558: conditional binary operator expected
/home/travis/.travis/job_stages: line 558: expected `)'
/home/travis/.travis/job_stages: line 558: syntax error near `-a'
/home/travis/.travis/job_stages: line 558: `  if [[ ("$TRAVIS_OS_NAME = linux" -a "$CC = gcc"'
/home/travis/.travis/job_stages: line 558: conditional binary operator expected
/home/travis/.travis/job_stages: line 558: expected `)'
```

leandro-lucarella-sociomantic replied:
Really? that's messed up. I sometimes don't understand what is Travis trying to do...

andreas-bok-sociomantic replied:
It is quite strange yes. The file with `-a` syntax passes the travis' online linter. 
I found the fix by trial and error. Would setting verbose flag for the shell give more insight into the mysterious ways of travis?

leandro-lucarella-sociomantic replied:
Yeah, usually adding `set -uxe` is a good idea, but with deployment scripts is also strange that you can only provide a one-liner, so you have to separate all commands via `;`.

andreas-bok-sociomantic replied:
Tried setting the the options with `bash -c "set -eux"` 
However  AFAICS it doesn't affect  the verbosity output compared to earlier builds `.travis`.
However we might set the options in v5.x.x which adds a separate bash script. 
Did experience though that the build errored due to the extensive logging after trying `set -v`

leandro-lucarella-sociomantic replied:
no, without the bash, just:
```yml
script: set -eux; beaver bintray ...
```

andreas-bok-sociomantic commented:
@leandro-lucarella-sociomantic 
`deploy` step is reached but fails with invalid cmd args since **BINTRAY_USER** is not defined. 
Does the values for **$BINTRAY_USER** and **$BINTRAY_KEY** correspond to github user and pass?

leandro-lucarella-sociomantic commented:
No, those are the credentials to upload to the sociomantic-tsunami bintray repos. If you reached that point, we need to continue the testing in the blessed GitHub repo and the real bintray repos. At least unless you want to create your own bintray user and debian repo and upload there. But IMHO is not worth the trouble.

andreas-bok-sociomantic commented:
> No, those are the credentials to upload to the sociomantic-tsunami bintray repos. 

Oki 
> If you reached that point, we need to continue the testing in the blessed GitHub repo and the real bintray repos

Once the syntax for the `deploy.condition` is settled I will do the patch release. And then the actual testing of the upload will then happen post v5.5.5

leandro-lucarella-sociomantic commented:
> Once the syntax for the deploy.condition is settled I will do the patch release. And then the actual testing of the upload will then happen post v5.5.5

The upload of the packages will happen as soon as you tag v5.5.5 and Travis tests for that tag pass :)

I suggest testing the package upload by making a v5.5.5-rc1 tag, this will upload the package as a pre-release, letting know the users this is not a final release (in the GitHub releases page you can also mark it as a pre-release).

andreas-bok-sociomantic commented:
Wrt to the `deploy.on.configtion`. Unless you insist I don't feel it is worth spending more time debugging the travis'

andreas-bok-sociomantic committed:
Bugfix: .travis deploy step

Wrong path to deb package output
Fix parsing error for deploy.on.condition
Add beaver bin folder to PATH env variable
Fix formatting of env.global setting

andreas-bok-sociomantic commented:
Squashed commits

leandro-lucarella-sociomantic commented:
> configtion

Condition I presume? Yes, move forward with it, I was just curious!

andreas-bok-sociomantic commented:
Unblocked

leandro-lucarella-sociomantic commented on .travis.yml:
```diff
@@ -22,10 +22,13 @@ services:
+    - MYSQL_SERVER=127.0.0.1 \
+      MYSQL_PASSWORD='' \
+      MYSQL_USER=root \
+      SKIP_TEST_ON_ERROR=0
```
Why not using one item per variable now? That format looks a bit strange.

andreas-bok-sociomantic replied:
Sure

leandro-lucarella-sociomantic reviewed (approved):
Other than the minor comment, LGTM

andreas-bok-sociomantic commented:
Updated based on comments from @leandro-lucarella-sociomantic

andreas-bok-sociomantic commented:
Ah

leandro-lucarella-sociomantic commented:
It still LGTM. Feel free to auto-merge unless you expect someone else to review this.

andreas-bok-sociomantic commented:
Oki, I will just test the `set -eux` before merging

andreas-bok-sociomantic commented:
> Oki, I will just test the set -eux before merging

Didn't see any change on the build log. I will merge it without the `set -eux`, but keep it in mind for the changes in `v5.x.x`. Tx for the reviews

andreas-bok-sociomantic commented:
Just a question, should it be released release candidate, ie. `v5.5.5-rc1`?

leandro-lucarella-sociomantic commented:
I would do that, yes. If all goes well we can tag the same commit as final, if not we add more commits to fix the problems before the final v5.5.5 release (and then merge to the other branches and release those too).

leandro-lucarella-sociomantic commented:
> Didn't see any change on the build log. I will merge it without the set -eux, but keep it in mind for the changes in v5.x.x. Tx for the reviews

The change should be there but minimal, only a `+beaver bintray ...` should be added to the output.

andreas-bok-sociomantic commented:
> The change should be there but minimal,

Ah ok

andreas-bok-sociomantic commented:
> . If all goes well we can tag the same commit as final, if not we add more commits to fix the problems before the final v5.5.5 release 

All went well. So I will do the patch release with the tag `v5.5.5`, right? OAIMS
